### PR TITLE
[quant] Make observer fields as paramteres instead of buffers

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45054 [quant] Make observer fields as paramteres instead of buffers**

Summary:
Currently the observer stats don't show up as a part of named_parameters in the module.
The difference between paramteres and bufers is that buffers do not update the gradient at every step.
We can get around this by specifying requires_grad=False for the parameters

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23810726](https://our.internmc.facebook.com/intern/diff/D23810726)

### CODE REVIEW

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -586,8 +586,8 @@ def __init__(self, ch_axis=0, dtype=torch.quint8,
                                                        quant_min=quant_min,
                                                        quant_max=quant_max)
         self.ch_axis = ch_axis
-        self.register_buffer('min_vals', torch.tensor([]))
-        self.register_buffer('max_vals', torch.tensor([]))
+        self.min_vals = Parameter(torch.tensor([]), requires_grad=False)
```
Comment (raghuramank100) [inline torch/quantization/observer.py:589]: This has implications on distributed training. When these are buffers, all replicas will use the same values as the master (broadcast), which is needed during training, so that all replicas use the same scale and zero-point for fake-quant. When they are parameters, they are not synchronized, which can cause an accuracy drop.

Comment (supriyar) [reply torch/quantization/observer.py:589]: I see, is this true even if we set `requires_grad=False` on the parameters?

**Comment by vkuzo** [comment]
> Currently the observer stats don't show up as a part of named_parameters in the module.

what would the end goal be, would it be for printing things during debugging, saving things, something else?  IMO if there is a way to achieve the end goal but keep these as buffers (since they don't need grad), that could be more inline with how PyTorch expects things to work.

**Comment by supriyar** [comment]
Closing this as we will figure out an alternate way to save the observer stats.
